rmgpy/molecule/molecule.py

```diff
@@ -1595,4 +1595,38 @@ def toGroup(self):
         
         return group
 
-
+    def getNthNeighbor(self, startingAtoms, n, blacklist):
+        '''
+        Recursively get the Nth nonHydrogen neighbors of the startingAtoms.
+        StartingAtoms should be a list.
+        Return a list.
+        '''
+        neighbors = []
+        for atom in startingAtoms:
+            for neighbor in self.getBonds(atom):
```

Comment:
Thanks for the comment 😃. Take a shot with 957480c384dce6ab14cf51b7512c7db226618966 by separating the double layer 'for' loop into two individual loops. I think speed is not an issue for the getNthNeighbor method though. But it might be an issue for this branch. Because the descendTree method will be called much more frequently than that in the master branch.